src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -624,6 +720,250 @@ namespace OpenMS
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `h` (the amplitude).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_h(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `mu` (the mean).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_mu(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `sigma` (the standard deviation).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_sigma(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute the cost given by the partial derivative of the loss function E,
+      with respect to `tau` (the exponent relaxation time).
+
+      Needed by the gradient descent algorithm.
+
+      @param[in] xs Positions
+      @param[in] ys Intensities
+      @param[in] h Amplitude
+      @param[in] mu Mean
+      @param[in] sigma Standard deviation
+      @param[in] tau Exponent relaxation time
+
+      @return The computed cost
+    */
+    double E_wrt_tau(
+      const std::vector<double>& xs,
+      const std::vector<double>& ys,
+      const double h,
+      const double mu,
+      const double sigma,
+      const double tau
+    ) const;
+
+    /**
+      @brief Compute EMG's z parameter
+
+      The value of z decides which formula is to be used during EMG function computation.
+      The interested ranges for z are:
```

Comment:
Instead of "The interested ranges for z are" use "Z values in the following ranges will each use a different EMG formula to avoid numerical instability and potential numerical overflow"